internal/lib/container_server.go

```diff
@@ -441,6 +452,12 @@ func (c *ContainerServer) LoadContainer(id string) error {
 	if err := json.Unmarshal(config, &m); err != nil {
 		return err
 	}
+
+	// Do not interact with containers of others
+	if manager, ok := m.Annotations[annotations.ContainerManager]; ok && manager != ContainerManagerCRIO {
+		return ErrIsNonCrioContainer
+	}
+
```

Comment:
in #2882 there was a check like:

```
return md.PodName != "" && md.PodID != ""
```

Is that worth doing here too?

Reply:
Yeah I think doing both. If we're restoring from an older version we still want to restore our crio containers that don't yet have it. We can deprecate the Pod{Name,ID} check in 1.17

Reply:
I agree, I added it inspired by https://github.com/cri-o/cri-o/pull/2882